### PR TITLE
[KYUUBI #4530]  [AUTHZ] Support non-English chars for MASK, MASK_SHOW_FIRST_4, and MASK_SHOW_FIRST_4

### PR DESCRIPTION
**Pull request by huangzhir**
### _Why are the changes needed?_
To fix https://github.com/apache/kyuubi/issues/4530.
1. The reason for issue https://github.com/apache/kyuubi/issues/4530  is that MASK_SHOW_FIRST_4 and MASK_SHOW_LAST_4 mask types are currently implemented using the regexp_replace method, which only replaces English letters and digits, but ignores other languages, such as Chinese.
2. To fix this issue, I modified the regexp_replace method to replace no-english characters to 'U' letters, so they will also be masked properly.

### _How was this patch tested?_

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @bowenliang123 @pan3793

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala
Comment (bowenliang123) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala:93]: Any reference for the starting and ending unicode for Chinese chars?

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala
Comment (bowenliang123) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala:93]: Chinese chars should not be part of upper chars. Use a separate replacer for it.

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala
Comment (bowenliang123) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala:91]: Use a separate ut cases for testing this `regexp_replace` method.

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala
Comment (bowenliang123) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala:397]: Add comments to explain what and why to use this unicodes in tests. Source code is in UTF-8 encoding and good to have Chinese chars.

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala
Comment (bowenliang123) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala:345]: Remove the unessary blank line.

**Review by bowenliang123 — CHANGES_REQUESTED** [review]
I have concerns in what to fix in this PR.
- What to do if expand the use range from original English chars to other Unicode chars, not limited to Chinese
- Use separate ut cases for testing `regexp_replace` method solely , for all the supposed supporting use ranges  
- Fix the conflicts

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala
Comment (huangzhir) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala]: This is because of Scala code style checks, which do not allow Chinese characters. Using Chinese characters may cause code style check errors, even though the code can still run.
image

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala
Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala:136]: so why `\u9fa5` used here?

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala
Comment (bowenliang123) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala:84]: I would say it's worse for understanding now.

**Comment by bowenliang123** [comment]
Overall, it is not in my favor to fix "Chinese" chars.
It would suggest we move one step back to discuss what range of the chars to fix (what about other languages or even other unicode chars) , what is the relation between `regrex_replace` to the purpose, and what possible solution in Spark or in Ranger (like how they do in Ranger's Hive plugin).

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala
Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala]: No. Source code do support utf8 chars including Chinese. Also, you would prefer reformatting tool for lining and consider inline-method comments. Comments are meant to make the case readable and easy to understand, especially you don't have exact Chinese words in the sql.

**Comment by bowenliang123** [comment]
I think we should follow the same making rule implementation jas in Ranger's DataMasker. And just mask all the chars into `"x"` without classifying in upper/lower/digits or any type. @yaooqinn @huangzhir 

（https://github.com/apache/ranger/blob/000db1cbeb0f8fed1e02e88edbdc80386023b515/plugin-nestedstructure/src/main/java/org/apache/ranger/authorization/nestedstructure/authorizer/DataMasker.java#L250）, 

```
    private static String showFirstFour(String value){
        int length = StringUtils.length(value);

        return length <= 4 ? value : value.substring(0, 4) + StringUtils.repeat("x", length - 4);
    }
```

**Comment by huangzhir** [comment]
Let me summarize how this issue came about and how Hive, Spark, and Trino handle it.

Hive's data masking is implemented using the functions mask({col}), mask_show_last_n({col}, 4, 'x', 'x', 'x', -1, '1'), and mask_show_first_n({col}, 4, 'x', 'x', 'x', -1, '1') (see https://github.com/apache/ranger/blob/7f5b82bff2df72f20f5c41ba095406d354f8acf0/agents-common/src/main/resources/service-defs/ranger-servicedef-hive.json#L387). 
```json 
                     {
				"itemId": 3,
				"name": "MASK_SHOW_FIRST_4",
				"label": "Partial mask: show first 4",
				"description": "Show first 4 characters; replace rest with 'x'",
				"transformer": "mask_show_first_n({col}, 4, 'x', 'x', 'x', -1, '1')"
			}
```
However, the implementation in the code ignores non-English character sets and simply returns the original data (see https://github.com/apache/hive/blob/7b3ecf617a6d46f48a3b6f77e0339fd4ad95a420/ql/src/java/org/apache/hadoop/hive/ql/udf/generic/GenericUDFMask.java#L262).
```java 
      default:
        if(maskedOtherChar != UNMASKED_VAL) {
          return maskedOtherChar;
        }
        break;
    }
```

Regarding the related mask functions in Spark, there was a JIRA ticket (https://issues.apache.org/jira/browse/SPARK-23901) to add mask-related functions, but it was later decided that this was not a universal method, so the code was rolled back, and Spark does not currently have an implementation of mask-related functions.

Trino also does not implement these mask-related functions, but instead uses the regexp_replace function for data masking. However, Trino's regexp_replace function supports lambda expressions (see https://github.com/apache/ranger/blob/a0224b2fdef999b3e23e2374080df94bf38557a4/agents-common/src/main/resources/service-defs/ranger-servicedef-trino.json#L393).
```json
{
        "itemId": 2,
        "name": "MASK_SHOW_LAST_4",
        "label": "Partial mask: show last 4",
        "description": "Show last 4 characters; replace rest with 'X'",
        "transformer": "cast(regexp_replace({col}, '(.*)(.{4}$)', x -> regexp_replace(x[1], '.', 'X') || x[2]) as {type})"
      }
```

**Comment by huangzhir** [comment]
> I think we should follow the same making rule implementation jas in Ranger's DataMasker. And just mask all the chars into `"x"` without classifying in upper/lower/digits or any other language type. @yaooqinn @huangzhir
> 
> （[https://github.com/apache/ranger/blob/000db1cbeb0f8fed1e02e88edbdc80386023b515/plugin-nestedstructure/src/main/java/org/apache/ranger/authorization/nestedstructure/authorizer/DataMasker.java#L250）](https://github.com/apache/ranger/blob/000db1cbeb0f8fed1e02e88edbdc80386023b515/plugin-nestedstructure/src/main/java/org/apache/ranger/authorization/nestedstructure/authorizer/DataMasker.java#L250%EF%BC%89),
> 
> ```
>     private static String showFirstFour(String value){
>         int length = StringUtils.length(value);
> 
>         return length <= 4 ? value : value.substring(0, 4) + StringUtils.repeat("x", length - 4);
>     }
> ```

If we don't refer to the implementations of Hive and Trino and there are no legacy issues, the natural way to implement data masking is to directly replace characters with "x". This PR is just a patch that fixes Chinese data masking and maintains some compatibility with Hive.

**Comment by bowenliang123** [comment]
This PR is not quite persuasive to me for now.
I would like to see,
1. an explanation and implementation of different kinds of unicode chars. AndAgain Chinese is just one type of them, so PR 
2. not sure relation between unicode chars and utf8 chars, whether in stored data itself or inside Spark engine
2. short of e2e test, to masking data in data in table 
3. short of convincing test to test each case separately , all the ut mixed together , and part of tests limited to 3.2+

If this PR keeps to fix Chinese chars only (and only within the basic unicode range), I think I would not to pass it.
And I would like to hear any idea from  @yaooqinn  @pan3793 @Yikf @zhouyifan279

**Comment by huangzhir** [comment]
I think we have the following solutions::
1. Make no changes and keep the behavior the same as Hive. (Convert uppercase to 'X', lowercase to 'x', and number to 'n')
2. Add a solution to address the issue of Chinese character obfuscation. (Convert Chinese characters to '密')
3. Replace each character set in Unicode with a different character.
4. Replace all characters except English letters and numbers with a single character. (Add replacement for other characters, e.g., convert all other characters to 'U')
5. Treat all characters in the same way. (Convert all characters to 'X')
6. Other solutions are welcome.

I am willing to work towards everyone's ideas to solve this problem.

**Comment by yaooqinn** [comment]
> Add replacement for other characters, e.g., convert all other characters to 'U'

+1

**Comment by huangzhir** [comment]
hello ,I'd like to request your help to review my PR. Based on @bowenliang123 @yaooqinn  your feedback, I've made some changes to the code and re-submitted the PR. The changes include:

Replacing all non-English letters, numbers, and spaces with the letter 'U'. This will ensure that all Unicode character sets for all languages are properly masked.
Adding unit tests specifically for regexp_replace, maskShowFirst4, and maskShowLast4.

@bowenliang123 Could you please review my changes and let me know if there's anything else I can improve on? Thank you

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala:128]: not necessary, you can add tests using sql queries

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala:149]: seems not necessary changes

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala
Comment (huangzhir) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala]: Thank you for your feedback. I have made the changes you suggested. Please let me know if there is anything else I can do to improve this PR.

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala:264]: hi @huangzhir, these tests are not verifying any code we changed. please use a normal query like `select col0 from t1`, where `t1.col0` has a masker configuration with show_xxx_4. It shall be done by the masking rule automatically instead of what we do here manually

Comment (yaooqinn) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala:264]: you can refer to other tests in this test base

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala
Comment (bowenliang123) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala:139]: The blank space should be treated as the common chars.

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala
Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala]: Maybe update the PR title too ?

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala
Comment (huangzhir) [reply extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPlugin.scala]: Thank you for your feedback. I have made the changes you suggested.

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala:279]: Can we move the data preparation to `setup`? Because it may become flaky when we add new tests in the future

**Comment by yaooqinn** [comment]
The overall change looks good to me, except for minor revisions.

**Comment by huangzhir** [comment]
there  are 2 checks were not successful. 
```
Error:  Plugin org.antlr:antlr4-maven-plugin:4.9.3 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.antlr:antlr4-maven-plugin:jar:4.9.3: The following artifacts could not be resolved: org.antlr:antlr4-maven-plugin:pom:4.9.3 (absent): Could not transfer artifact org.antlr:antlr4-maven-plugin:pom:4.9.3 from/to gcs-maven-central-mirror (https://maven-central-asia.storage-download.googleapis.com/maven2/): Connection timed out (Read failed) -> [Help 1]
```
```
Error:  Failed to execute goal on project kyuubi-spark-sql-engine_2.12: Could not resolve dependencies for project org.apache.kyuubi:kyuubi-spark-sql-engine_2.12:jar:1.8.0-SNAPSHOT: The following artifacts could not be resolved: org.apache.spark:spark-streaming_2.12:jar:3.2.3 (absent): Could not transfer artifact org.apache.spark:spark-streaming_2.12:jar:3.2.3 from/to gcs-maven-central-mirror (https://maven-central-asia.storage-download.googleapis.com/maven2/): Connection timed out (Read failed) -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal on project kyuubi-spark-sql-engine_2.12: Could not resolve dependencies for project org.apache.kyuubi:kyuubi-spark-sql-engine_2.12:jar:1.8.0-SNAPSHOT: The following artifacts could not be resolved: org.apache.spark:spark-streaming_2.12:jar:3.2.3 (absent): Could not transfer artifact org.apache.spark:spark-streaming_2.12:jar:3.2.3 from/to gcs-maven-central-mirror (https://maven-central-asia.storage-download.googleapis.com/maven2/): Connection timed out (Read failed)
```
These two errors seem to be caused by a network problem with connecting to https://maven-central-asia.storage-download.googleapis.com/maven2. After looking at other PRs, it seems that this issue is not unique.

**Comment by pan3793** [comment]
Yea, the Google Maven Central mirror is not stable these days, I re-trigger the failed Job.

**Comment by huangzhir** [comment]
```
Error:  Plugin org.antlr:antlr4-maven-plugin:4.9.3 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.antlr:antlr4-maven-plugin:jar:4.9.3: The following artifacts could not be resolved: org.antlr:antlr4-maven-plugin:pom:4.9.3 (absent): Could not transfer artifact org.antlr:antlr4-maven-plugin:pom:4.9.3 from/to gcs-maven-central-mirror (https://maven-central-asia.storage-download.googleapis.com/maven2/): Connection timed out (Read failed) -> [Help 1]
org.apache.maven.plugin.PluginResolutionException: Plugin org.antlr:antlr4-maven-plugin:4.9.3 or one of its dependencies could not be resolved: Failed to read artifact descriptor for org.antlr:antlr4-maven-plugin:jar:4.9.3
```
HI  @pan3793 ,There is still a check  not successful, and it seems to be a network issue. Could you please help to re-trigger the failed job again?

**Comment by pan3793** [comment]
@huangzhir seems there is an un-addressed comment, and you can simply ignore the error caused by the network, the committer will check it before merging

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala
Comment (huangzhir) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala:279]: I originally wanted to put it in the setup method, but putting it there caused several old test cases to fail. We need to modify the old test cases, such as adding a where clause in the old test cases. Do you mind this implementation approach?

Comment (yaooqinn) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala:279]: revising the old ones is OK with me.

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPluginSuite.scala
Comment (bowenliang123) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPluginSuite.scala:60]: I'm not sure about what's tested here. can we use plain value for assertion ?

**Review by bowenliang123 — APPROVED** [review]
LGTM overall. Minor comments.

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPluginSuite.scala
Comment (huangzhir) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPluginSuite.scala:60]: This is a test of the correctness of the getMaskingExpr method for the MASK_SHOW_LAST_4 rule, referring to the assert test MASK rule

Comment (huangzhir) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPluginSuite.scala:60]: > can we use plain value for assertion

I'm sorry, I didn't understand the meaning of this sentence.

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala
Comment (huangzhir) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala]: Thank you for your feedback. I have made the changes you suggested. Move the data preparation to setup，some tests were modified due to changes in the data.

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPluginSuite.scala
Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPluginSuite.scala:60]: I don't understand this test assertion.

**Review by bowenliang123 — CHANGES_REQUESTED** [review]
Why is the filter condition `where key=1` added everywhere? Prefer not to break the existing regression tests, as it may change the behavior and plan for the SQL. And if you are determine to do so, explain the reasons and use separate tests.

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala
Comment (huangzhir) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala]: > Why is the filter condition `where key=1` added everywhere

 Based on @yaooqinn  suggestion, the data  preparation was moved to the setup method, which caused some changes in the data. The original test cases need to be modified by adding a where condition to keep searching for the original data.

Comment (huangzhir) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala]: > Why is the filter condition where key=1 added everywhere

Based on @yaooqinn suggestion, the data preparation was moved to the setup method, which caused some changes in the data. The original test cases need to be modified by adding a where condition to keep searching for the original data.

**Comment by huangzhir** [comment]
> Why is the filter condition where key=1 added everywhere

Based on @yaooqinn suggestion, the data preparation was moved to the setup method, which caused some changes in the data. The original test cases need to be modified by adding a where condition to keep searching for the original data. Maintain the integrity of the old test results.

**Review by yaooqinn — APPROVED** [review]

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPluginSuite.scala
Comment (huangzhir) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPluginSuite.scala:60]: Originally, we used a simple 'contains' method to check. I think this can be improved by using '===' directly, which can more accurately determine whether the method of generating MASK_SHOW_LAST_4 based on this rule is correct or not.

Comment (bowenliang123) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPluginSuite.scala:60]: From my view, I would like to see a clear assertion with the pair of a concrete expected value  and a code to run. In current implementation , both sides on the assertion contain evaluation.
WDYT ? @yaooqinn

Comment (yaooqinn) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/SparkRangerAdminPluginSuite.scala:60]: I'm OK with the current implementation.

These tests are unimportant as we have e2e test cases. They are introduced when the data masking is not fully implemented. And at that time, I want to ensure the `getMaskingExpr` works well.

> both sides on the assertion contain evaluation.

I do not get your point. The author only uses '+' for newline.

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala:131]: nit: 2 spaces

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala:141]: ditto

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala:189]: ditto

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala
Comment (huangzhir) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala]: Fixed, thanks!

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala
Comment (huangzhir) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala]: Fixed, thanks!

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala
Comment (huangzhir) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/datamasking/DataMaskingTestBase.scala]: Fixed, thanks!

**Review by bowenliang123 — APPROVED** [review]

**Comment by yaooqinn** [comment]
Thanks, @huangzhir, for the hard-working and patience all the way thru.

Merged to master.

**Comment by huangzhir** [comment]
You're welcome! It was my pleasure to contribute to this project.

**Pull request closed without merging**
